src/main/java/org/spongepowered/api/Server.java

```diff
@@ -86,6 +89,21 @@
     Optional<Player> getPlayer(String name);
 
     /**
+     * Gets a {@link Player} by their name, as seen by the
+     * provided viewer.
+     *
+     * <p>This only works for online players.</p>
+     *
+     * <p><b>Note: Do not use names for persistent storage, the
+     * Notch of today may not be the Notch of yesterday.</b></p>
+     *
+     * @param name The name to get the player from
+     * @param viewer The viewer of the request
+     * @return {@link Player} or Optional.empty() if not found
+     */
+    Optional<Player> getPlayer(String name, @Nullable ContextViewer viewer);
```

Comment:
I don't think it makes sense for the entire `Player` to change depending on the `ContextViewer`, since this is only going to affect Data.

Reply:
This makes sense to me, but would we need a similar method for uuids as seen by X. Are uuids ever exposed to the client? Or only names and skin data?